executor/include/executor/default/omp_executor.hpp

```diff
@@ -36,13 +36,27 @@ struct omp_executor {
     int idx;
   };
 
-  template <typename Executor, instance_of_base<omp_executor, Executor> = 0>
+  std::size_t max_threads;
+
+  omp_executor() {
+#ifdef _OPENMP
+    this->max_threads = omp_get_max_threads();
+#endif
+  }
+
+  omp_executor(std::size_t max_threads): max_threads(max_threads) {
+#ifdef _OPENMP
+    if (!this->max_threads) this->max_threads = omp_get_max_threads();
+#endif
```

Comment:
`ifndef _OPENMP`, should `max_threads` should be forced 0?

Reply:
I don't think so, since it is just to specify the maximum number of threads, so its the responsibility of the user to ensure OMP is available if it isn't he will anyway not be able to call using the executor.